tools/rc/ResourceCompiler.cpp

```diff
@@ -411,36 +413,54 @@ int main(int argc, char** argv)
     }
     return_code = EXIT_FAILURE;
   }
-  
-  option::Option* bundleFileOpt = options[BUNDLEFILE];
-  if (bundleFileOpt && bundleFileOpt->count() > 1 )
+
+  // multiple specifications of the following args are illegal.
+  auto check_multiple_args = [&options, &return_code](OptionIndex optionidx, const std::string& arg)
   {
-    std::cerr << "(--bundle-file | -b) appear multiple times in the arguments. Check usage." << std::endl;
-    return_code = EXIT_FAILURE;
-  }
-  
-  option::Option* outFileOpt = options[OUTFILE];
-  if (outFileOpt && outFileOpt->count() > 1 )
+    option::Option* opt = options[optionidx];
+    if (opt && opt->count() > 1 )
+    {
+      std::cerr << arg << " appears multiple times in the arguments. Check usage." << std::endl;
+      return_code = EXIT_FAILURE;
+    }
+  };
+  check_multiple_args(BUNDLEFILE, "(--bundle-file | -b)");
+  check_multiple_args(OUTFILE, "(--out-file | -o)");
+  check_multiple_args(BUNDLENAME, "(--bundle-name | -n)");
+  check_multiple_args(MANIFESTADD, "(--manifest-add | -m)");
+
+  if (argc == 0 || options[HELP])
   {
-    std::cerr << "(--out-file | -o) appear multiple times in the arguments. Check usage." << std::endl;
-    return_code = EXIT_FAILURE;
+    option::printUsage(std::clog, usage);
+    return return_code;
   }
   
+  option::Option* bundleFileOpt = options[BUNDLEFILE];
+  option::Option* outFileOpt = options[OUTFILE];
   if (!bundleFileOpt && !outFileOpt)
   {
-    std::cerr << "At least one of the options (--bundle-file | --out-file) is required." << std::endl;
+    std::cerr << "At least one of the options (--bundle-file | --out-file) is required. Check usage." << std::endl;
     return_code = EXIT_FAILURE;
   }
-  
-  if (argc == 0 || options[HELP] || return_code == EXIT_FAILURE)
+
+  if ((options[MANIFESTADD] || options[RESADD]) && !options[BUNDLENAME])
   {
-    option::printUsage(std::clog, usage);
-    return return_code;
+      std::cerr << "if either --manifest-add or --res-add option is provided, --bundle-name option must be provided." << std::endl;
```

Comment:
nit-pick on grammar: Start sentence with capital letter and remove redundant word "option".

> ```If either --manifest-add or --res-add option is provided, --bundle-name must be provided.```